src/components/tabController2/TabBar.tsx

```diff
@@ -0,0 +1,321 @@
+    labelStyle,
+    selectedLabelStyle,
+    labelColor,
+    selectedLabelColor,
+    uppercase,
+    iconColor,
+    selectedIconColor,
+    activeBackgroundColor,
+    backgroundColor,
+    containerWidth: propsContainerWidth,
+    centerSelected,
+    containerStyle,
+    testID,
+    children: propsChildren
+  } = props;
+
+  const context = useContext(TabBarContext);
+  const {itemStates, items: contextItems, currentPage, targetPage, selectedIndex} = context;
+
+  const children = useRef<Props['children']>(_.filter(propsChildren, (child: ChildProps) => !!child));
+
+  const containerWidth: number = useMemo(() => {
+    return propsContainerWidth || Constants.screenWidth;
+  }, [propsContainerWidth]);
+
+  const items = useMemo(() => {
+    return contextItems || propsItems;
+  }, [contextItems, propsItems]);
+
+  const itemsCount = useRef<number>(items ? _.size(items) : React.Children.count(children.current));
+
+  const {
+    scrollViewRef: tabBar,
+    onItemLayout,
+    itemsWidthsAnimated,
+    itemsOffsetsAnimated,
+    // itemsWidths,
+    // itemsOffsets,
+    focusIndex,
+    onContentSizeChange,
+    onLayout
+  } = useScrollToItem({
+    itemsCount: itemsCount.current,
+    selectedIndex,
+    offsetType: centerSelected ? useScrollToItem.offsetType.CENTER : useScrollToItem.offsetType.DYNAMIC
+  });
+
+  useAnimatedReaction(() => {
+    return Math.round(currentPage.value);
+  },
+  (currIndex, prevIndex) => {
+    if (currIndex !== prevIndex) {
+      runOnJS(focusIndex)(currIndex);
+    }
+  });
+
+  const _renderTabBarItems = useMemo((): ReactNode => {
+    return _.map(items, (item, index) => {
+      return (
+        <TabBarItem
+          labelColor={labelColor}
+          selectedLabelColor={selectedLabelColor}
+          labelStyle={labelStyle}
+          selectedLabelStyle={selectedLabelStyle}
+          uppercase={uppercase}
+          iconColor={iconColor}
+          selectedIconColor={selectedIconColor}
+          activeBackgroundColor={activeBackgroundColor}
+          key={item.label}
+          {...item}
+          {...context}
+          index={index}
+          // @ts-expect-error
+          state={itemStates[index]}
+          onLayout={onItemLayout}
+        />
+      );
+    });
+  }, [
+    items,
+    labelColor,
+    selectedLabelColor,
+    labelStyle,
+    selectedLabelStyle,
+    uppercase,
+    iconColor,
+    selectedIconColor,
+    activeBackgroundColor,
+    itemStates,
+    centerSelected,
+    onItemLayout
+  ]);
+
+  const _indicatorTransitionStyle = useAnimatedStyle(() => {
+    const value = targetPage.value;
+    const width = interpolate(value,
+      itemsWidthsAnimated.value.map((_v: number, i: number) => i),
+      itemsWidthsAnimated.value.map((v: number) => v - 2 * INDICATOR_INSET));
+
+    const left = interpolate(value,
+      itemsOffsetsAnimated.value.map((_v: any, i: number) => i),
+      itemsOffsetsAnimated.value);
+
+    return {
+      width,
+      left
+    };
+  });
```

Comment:
I think that's because they connected in the native side. 
If I'm not wrong, the idea here is to pass JS values (props, state) that affect the animateStyle hook and not sharedValues